ql/src/java/org/apache/hadoop/hive/ql/parse/RewriteSemanticAnalyzer.java

```diff
@@ -409,7 +409,14 @@ private boolean isTargetTable(Entity entity, Table targetTable) {
      * is this the right way to compare?  Should it just compare paths?
      * equals() impl looks heavy weight
      */
-    return targetTable.equals(entity.getTable());
+    long targetWriteId = targetTable.getTTable().getWriteId();
+    long entityWriteId = entity.getTable().getTTable().getWriteId();
+    targetTable.getTTable().setWriteId(0L);
+    entity.getTable().getTTable().setWriteId(0L);
+    boolean result = targetTable.equals(entity.getTable());
+    targetTable.getTTable().setWriteId(targetWriteId);
+    entity.getTable().getTTable().setWriteId(entityWriteId);
+    return result;
```

Comment:
I would prefer to have a new equalsWithIgnoreWriteId - or whatever better named method in Table.

Reply:
Makes sense. This will show up in the next commit.